packages/core/src/routes/interaction/types/guard.ts

```diff
@@ -6,16 +6,43 @@ import type {
   PhonePasswordPayload,
   PhonePasscodePayload,
 } from '@logto/schemas';
-import { eventGuard, profileGuard, identifierGuard } from '@logto/schemas';
+import {
+  socialConnectorPayloadGuard,
+  emailPasscodePayloadGuard,
+  phonePasscodePayloadGuard,
+  eventGuard,
+  profileGuard,
+  identifierGuard,
+} from '@logto/schemas';
 import { z } from 'zod';
 
 // Interaction Route Guard
-export const interactionPayloadGuard = z.object({
-  event: eventGuard.optional(),
+const forgotPasswordInteractionPayloadGuard = z.object({
+  event: z.literal('forgot-password'),
+  identifier: z.union([emailPasscodePayloadGuard, phonePasscodePayloadGuard]).optional(),
+  profile: profileGuard.pick({ password: true }).optional(),
+});
+
+const registerInteractionPayloadGuard = z.object({
+  event: z.literal('register'),
+  identifier: z
+    .union([emailPasscodePayloadGuard, phonePasscodePayloadGuard, socialConnectorPayloadGuard])
```

Comment:
password identifiers can not be used to register?

Reply:
email and phone identifier must be passcode verified before registeation. Password is allowed in the profile.